packages/zk-certificates/contracts/ZkKYC.sol

```diff
@@ -89,6 +89,32 @@ contract ZkKYC is Ownable {
         uint[2][2] memory b,
         uint[2] memory c,
         uint[] memory input
+    ) public view returns (bool) {
+            // dev note: if we ever use proof hash, make sure to pay attention to this truncation to uint160 as it can violate uniqueness
+        require(
+            tx.origin == address(uint160(input[INDEX_USER_ADDRESS])),
+            'transaction submitter is not authorized to use this proof'
+        );
+        return _verifyProof(a, b, c, input);
+    }
+
+    function verifyProof2(
+        uint[2] memory a,
+        uint[2][2] memory b,
+        uint[2] memory c,
+        uint[] memory input
+    ) public view returns (bool) {
+        return _verifyProof(a, b, c, input);
```

Comment:
in the faucet proof utilization, we need anyone to be able to submit proof but the fund can only be sent to an approved address. 
1) verifyProof: check that submitter is the authorized address
2) verifyProof2: doesn't care what the submitter is, but it's the burden of the Dapp to check what to do with the authorized address, in this case the fund receiver.
I made the internal _verifyProof to be the code that is shared between verifyProof and verifyProof2, because otherwise there would be a lot of duplication.